cmd/cluster/aws/create.go

```diff
@@ -77,10 +85,89 @@ func applyPlatformSpecificsValues(ctx context.Context, exampleOptions *apifixtur
 	if err != nil {
 		return err
 	}
-	infraID := opts.InfraID
-
 	// Load or create infrastructure for the cluster
 	var infra *awsinfra.CreateInfraOutput
+	var iamInfo *awsinfra.CreateIAMOutput
+	infraID := opts.InfraID
+
+	if infraID == "" {
+		return ErrMsg("infra-id")
+	}
+	hci := hyperv1.HostedClusterInfrastructure{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      opts.InfraID,
+			Namespace: opts.Namespace,
+		},
+	}
+	if err := client.Get(ctx, crclient.ObjectKeyFromObject(&hci), &hci); err != nil {
+		if apierrors.IsNotFound(err) {
+			opts.Log.Info("Hosted cluster infrastructure not found, creating infrastructure with user input", "namespace", opts.Namespace, "name", opts.InfraID)
+		} else {
+			return err
+		}
+	} else {
+		opts.Log.Info("Found hosted cluster infrastructure", "namespace", opts.Namespace, "name", opts.InfraID)
+		if hci.Spec.SubDomain != opts.Name {
+			return fmt.Errorf(fmt.Sprintf("The HostedClusterInfrastructure.Spec.SubDomain must equal the cluster name\nCluster name: %s, Sub-Domain: %s", opts.Name, hci.Spec.SubDomain))
+		}
+		psStatus, infraStatus, iamStatus := hostedclusterinfrastructure.GetInfraIamConditionStatus(&hci)
```

Comment:
what is ps, can we use a more explicit Name?
Can we have all this validations in a unit tested validateHCI() function

Reply:
ProvideSecret.. ProviderSecretStatus seemed long. I made two changes.
1. Renamed the method to `GetHciConditionStatus`
2. I added a comment for the func, so if you mouse over it, it tells you all three status's that are returned.